src/transaction/types/input.rs

```diff
@@ -230,6 +230,16 @@ impl Input {
             _ => None,
         }
     }
+
+    #[cfg(feature = "std")]
+    pub fn is_predicate_owner_valid(owner: &Address, predicate: &[u8]) -> bool {
+        use crate::Contract;
+
+        // TODO use as no-std as soon as a no-std merkle backend is available
+        let root = Contract::root_from_code(predicate);
```

Comment:
Nit: 
I'd personally suggest something like:
```rust
let contract = Contract::from(predicate);
let root = contract.root();
...
```
where the `from` is provided by [this line](https://github.com/FuelLabs/fuel-tx/pull/128/files#diff-1da256e88678c7e356c71cc74bdc1bc0411aa5f9c04115869079cbe5c5ef6a98R116). This means you only have to add `root` to the public interface, and you can keep `root_from_code` part of the private implementation. Less public entry points means less functions to test.

Reply:
The `root_from_code` is available to allow the user to define roots without reallocating.

If we use a `Contract::from` every time, then we always end up making an extra heap allocation: https://github.com/FuelLabs/fuel-tx/blob/32945270813e75567c55ef417149745c9b66c158/src/contract.rs#L23